distsys/resources/tcpmailboxes.go

```diff
@@ -31,6 +31,36 @@ const (
 	tcpMailboxesConnectionDroppedRetryDelay = 50 * time.Millisecond // TODO: same
 )
 
+// TCPMailboxesArchetypeResourceMaker produces a distsys.ArchetypeResourceMaker for a collection of TCP mailboxes.
+// Each individual mailbox will match the following mapping macro, assuming exactly one process "reads" from it:
+//
+//     mapping macro TCPChannel {
+//       read {
+//           await Len($variable) > 0;
+//           with (msg = Head($variable)) {
+//               $variable := Tail($variable);
+//               yield msg;
+//           };
+//       }
+//
+//       write {
+//           await Len($variable) < BUFFER_SIZE;
+//           yield Append($variable, $value);
+//       }
+//     }
+//
+// As is shown above, each mailbox should be a fully reliable, ordered channel, which these resources approximate
```

Comment:
I'll need to read your document more closely... we'd also need to include property definitions or links, because I don't think these Go docs should require that writing as context, without at least linking it.